### PR TITLE
Update RepositoryServer CR Status field to add `metav1.Condition`

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview

This PR updates RepositoryServer CR Status fields, we are now using [https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition) instead of custom condition for RepositoryServer CR.

As a part of this PR we are also modifying the RepositoryServer CRD

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

NA

### CODE REVIEW

File: pkg/apis/cr/v1alpha1/repositoryserver_types.go [hunk 1]
```diff
@@ -117,48 +117,44 @@ type UserAccess struct {
...
-	RepositoryReady RepositoryServerConditionType = "RepositoryReady"
+	RepositoryReady string = "RepositoryReady"
 
 	// ServerInitialized means that the proxy server, that serves the repository, has been started
-	ServerInitialized RepositoryServerConditionType = "ServerInitialized"
+	ServerInitialized string = "ServerInitialized"
 
 	// ClientsInitialized indicates that the client users have been added or updated to the repository server
-	ClientsInitialized RepositoryServerConditionType = "ClientsInitialized"
+	ClientsInitialized string = "ClientsInitialized"
```
Comment (kale-amruta) [inline pkg/apis/cr/v1alpha1/repositoryserver_types.go:140]: we decided to change this to `ClientUserInitialized` right?

**Review by kale-amruta — COMMENTED** [review]
looks good to me with one minor comment

File: pkg/apis/cr/v1alpha1/repositoryserver_types.go (hunk 1 above)
Comment (akankshakumari393) [reply pkg/apis/cr/v1alpha1/repositoryserver_types.go]: Right. I missed it somehow. Have updated it now.

**Review by kale-amruta — APPROVED** [review]
lgtm

**Pull request merged**